src/VisualStudio/Core/Def/Implementation/ProjectSystem/CPS/IWorkspaceProjectContext.cs

```diff
@@ -12,8 +12,12 @@
 namespace Microsoft.VisualStudio.LanguageServices.ProjectSystem
 {
     /// <summary>
-    /// Project context to initialize properties and items of a Workspace project created with <see cref="IWorkspaceProjectContextFactory.CreateProjectContext(string, string, string, Guid, object, string)"/>. 
+    /// Project context to initialize properties and items of a Workspace project created with <see
+    /// cref="IWorkspaceProjectContextFactory.CreateProjectContextAsync"/>. 
     /// </summary>
+    /// <remarks>
+    /// <see cref="IDisposable.Dispose"/> is safe to call on instances of this type on any thread.
```

Comment:
❔ Can it be called while asynchronous operations are scheduled and/or in progress?

Reply:
IWorkspaceProjectContext.cs itself doesn't do anything asynchronously.  There is async work to create it, but once you have it, it's a sync object.

Reply:
And yeah we'll be putting more async operations onto it at some point; so we'll need to sort that out...later.

Reply:
Actually we should just make a proper ExternalAccess setup for CPS at that point.